src/compiler/binder.ts

```diff
@@ -1913,6 +1913,15 @@ namespace ts {
             }
         }
 
+        function checkStrictModeLabeledStatement(node: LabeledStatement) {
+            // Grammar checking for labeledStatement
+            if (inStrictMode && options.target! >= ScriptTarget.ES2015) {
+                if (isDeclarationStatement(node.statement) || isVariableStatement(node.statement)) {
+                    errorOnFirstToken(node.label, Diagnostics._0_declarations_cannot_be_labeled, formatSyntaxKind(node.statement.kind));

```

Comment:
I do not think we ever exposed these kinds in error messages..

What we have done is have a generic message, like `A label is not allowed here`, or `Invalid use of label`.

Reply:
the other alternative is to have a switch statement for the kind, e.g. `"function"` for `SyntaxKind.FunctonDeclaration`, but personally I do not think the value you get is worth the effort.